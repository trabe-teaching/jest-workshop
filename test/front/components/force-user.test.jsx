import React from "react";
import { shallow } from "enzyme";
import ForceUser from "components/force-user";

describe("front/components/force-user", () => {
  const user = {
    id: 100,
    name: "Luke Skywalker",
    type: "light",
    rank: "master",
    status: "alive",
    email: "luke@ach.to",
  };

  it("renders a force user", () => {
    const wrapper = shallow(<ForceUser user={user} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("actions", () => {
    ["Banish"].forEach(action => {
      it(`renders a force user that can ${action}`, () => {
        const handler = jest.fn();
        const preventDefault = jest.fn();

        const props = {
          user: {
            ...user,
            [`can${action}`]: true,
          },
          [`on${action}`]: handler,
        };

        const wrapper = shallow(<ForceUser {...props} />);
        expect(wrapper).toMatchSnapshot();

        wrapper.find(".button").simulate("click", { preventDefault });
        expect(preventDefault).toHaveBeenCalledTimes(1);
        expect(handler).toHaveBeenCalledTimes(1);
        expect(handler).toHaveBeenCalledWith(100);
      });
    });
  });
});
