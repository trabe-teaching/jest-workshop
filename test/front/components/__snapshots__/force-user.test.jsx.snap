// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`front/components/force-user actions renders a force user that can Banish 1`] = `
<div
  className="box force-user force-user--light force-user--alive"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <Avatar
        email="luke@ach.to"
      />
    </div>
    <div
      className="media-content"
    >
      <div
        className="content"
      >
        <p>
          <strong>
            Luke Skywalker
          </strong>
          <span
            className="tag is-gray is-pulled-right"
          >
            master
          </span>
        </p>
        <div
          className="field is-grouped"
        >
          <div
            className="control"
          >
            <a
              className="button is-small"
              href="#"
              onClick={[Function]}
            >
              Join the force
            </a>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`front/components/force-user renders a force user 1`] = `
<div
  className="box force-user force-user--light force-user--alive"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <Avatar
        email="luke@ach.to"
      />
    </div>
    <div
      className="media-content"
    >
      <div
        className="content"
      >
        <p>
          <strong>
            Luke Skywalker
          </strong>
          <span
            className="tag is-gray is-pulled-right"
          >
            master
          </span>
        </p>
        <div
          className="field is-grouped"
        />
      </div>
    </div>
  </article>
</div>
`;
